src/RichXMLParser.m

```diff
@@ -57,8 +57,15 @@ -(BOOL)parseRichXML:(NSData *)xmlData
         xmlDocument = [[NSXMLDocument alloc] initWithData:xmlData
                                                         options:NSXMLNodeOptionsNone
                                                         error:&error];
-        if (error) {
-            if (error.code == 73) return NO;
+        if (xmlDocument == nil && error != nil) {
+			if ([error.domain isEqualToString:NSXMLParserErrorDomain]) {
+				if (error.code == NSXMLParserGTRequiredError)
+					return NO;
+				if (error.code == NSXMLParserPrematureDocumentEndError)
+					return NO;
+			}
+			// Do we even want to try this?
```

Comment:
Especially, an attempt with the `NSXMLDocumentTidyXML` option enabled will recover from invalid characters in feeds, while `NSXMLNodeOptionsNone` would return an empty document.